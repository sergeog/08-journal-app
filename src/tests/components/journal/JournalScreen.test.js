import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';

import { JournalScreen } from '../../../components/journal/JournalScreen';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initialState = {
  auth: {
    name: 'Serge',
  },
  notes: {
    notes: [],
    active: null,
  },
};
const store = mockStore(initialState);

const wrapper = mount(
  <Provider store={store}>
    <JournalScreen />
  </Provider>
);

describe('Pruebas en <JournalScreen />', () => {
  test('debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe mostrar el componente <NothingSelected /> cuando no haya alguna nota activa', () => {
    const nothingSelectedTag = wrapper.find('NothingSelected');
    expect(nothingSelectedTag.exists()).toBe(true);
  });
});
