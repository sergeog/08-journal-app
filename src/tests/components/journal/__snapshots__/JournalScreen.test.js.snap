// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <JournalScreen /> debe mostrarse correctamente 1`] = `
<div
  className="journal__main-content animate__animated animate__fadeIn animate__faster"
>
  <aside
    className="journal__sidebar"
  >
    <div
      className="journal__sidebar-navbar"
    >
      <h3
        className="mt-5"
      >
        <i
          className="far fa-moon"
        />
        <span>
           
          Serge
        </span>
      </h3>
      <button
        className="btn"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
    <div
      className="journal__new-entry"
      onClick={[Function]}
    >
      <i
        className="far fa-calendar-plus fa-5x"
      />
      <p
        className="mt-5"
      >
        New Entry
      </p>
    </div>
    <div
      className="journal__entries"
    />
  </aside>
  <main>
    <div
      className="nothing__main-content"
    >
      <p>
        Select something
        <br />
        or create an entry!
      </p>
      <i
        className="far fa-star fa-4x mt-5"
      />
    </div>
  </main>
</div>
`;
